### PR TITLE
add --version flag

### LINKED ISSUE #228: conda-lock --version doesn't work
I'm used to being able to check version numbers with

```bash
conda-lock --version
```

but unfortunately I get

```
Error: No such option: --version
```

### PR DESCRIPTION
**Pull request by bollwyvl**
## References

- fixes #228

## Changes
- [x] adds `click.version_option`
- [x] sort, normalize, and deduplicate `requirements.txt` 
- [x] test

### CODE REVIEW

**Comment by bollwyvl** [comment]
Doesn't seem like the windows fail is related.

File: requirements.txt [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
-pydantic >=1.8.1
+ensureconda >=1.3
+jinja2
 poetry <1.2
+pydantic >=1.8.1
+pyyaml >= 5.1
+requests >=2
 ruamel.yaml
-poetry
+toml
```
Comment (bollwyvl) [inline requirements.txt:10]: should probably start using `tomli`...

Comment (mariusvniekerk) [reply requirements.txt:10]: What's the difference?

Comment (bollwyvl) [reply requirements.txt:10]: sorry, confused with `pytoml`, which is definitely deprecated.

https://github.com/uiri/toml hasn't had a substantive PR merged in some time, and many open issues, but is probably still fine. 

I guess the biggest thing in favor of `tomli` is it's just the parsing side, and is the basis for the upcoming `tomllib` in stdlib: https://peps.python.org/pep-0680/#rationale

Comment (mariusvniekerk) [reply requirements.txt:10]: If you want to make a or to shift to tomli (or I guess preferentially the stdlib one for py 3.11/12) that would be awesome.

**Pull request merged**